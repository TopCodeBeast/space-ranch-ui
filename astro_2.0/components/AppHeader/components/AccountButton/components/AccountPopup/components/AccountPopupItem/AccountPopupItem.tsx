import cn from 'classnames';
import Link from 'next/link';
import React, { FC } from 'react';

import { useIsHrefActive } from 'hooks/useIsHrefActive';

import styles from './AccountPopupItem.module.scss';

interface AccountPopupItemProps {
  href?: string;
  className?: string;
  onClick?: () => void;
}

export const AccountPopupItem: FC<AccountPopupItemProps> = ({
  href,
  className,
  children,
  onClick,
}) => {
  const isActive = useIsHrefActive(href);

  const content = (
    <div
      tabIndex={0}
      role="button"
      onClick={onClick}
      onKeyPress={onClick}
      className={cn(styles.root, className, {
        [styles.active]: isActive,
      })}
    >
      {children}
    </div>
  );

  if (!href) {
    return content;
  }

  return <Link href={href}>{content}</Link>;
};
