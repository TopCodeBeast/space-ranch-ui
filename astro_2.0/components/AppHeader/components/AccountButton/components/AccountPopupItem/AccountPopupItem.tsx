import cn from 'classnames';
import React, { FC } from 'react';

import styles from './AccountPopupItem.module.scss';

interface AccountPopupItemProps {
  className?: string;
  onClick?: () => void;
}

export const AccountPopupItem: FC<AccountPopupItemProps> = ({
  className,
  children,
  onClick,
}) => {
  return (
    <div
      tabIndex={0}
      role="button"
      onClick={onClick}
      onKeyPress={onClick}
      className={cn(styles.root, className)}
    >
      {children}
    </div>
  );
};
