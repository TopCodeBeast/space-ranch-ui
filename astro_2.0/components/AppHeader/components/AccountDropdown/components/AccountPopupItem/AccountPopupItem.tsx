import cn from 'classnames';
import React, { FC } from 'react';

import styles from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/AccountPopupItem/AccountPopupItem.module.scss';

interface AccountPopupItemProps {
  icon: React.ReactNode;
  classes?: {
    root?: string;
    content?: string;
  };

  onClick?: () => void;
}

export const AccountPopupItem: FC<AccountPopupItemProps> = ({
  classes = {},
  icon,
  children,
  onClick,
}) => {
  return (
    <div
      tabIndex={0}
      role="button"
      onClick={onClick}
      onKeyPress={onClick}
      className={cn(styles.root, classes.root)}
    >
      <div className={styles.icon}>{icon}</div>
      <div className={cn(styles.content, classes.content)}>{children}</div>
    </div>
  );
};
