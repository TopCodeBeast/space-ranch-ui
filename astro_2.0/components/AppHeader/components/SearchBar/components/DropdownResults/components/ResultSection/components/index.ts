export * from './SearchResultDaoCard';
export * from './SearchResultPeopleCard';
export * from './SearchResultProposalLine';
