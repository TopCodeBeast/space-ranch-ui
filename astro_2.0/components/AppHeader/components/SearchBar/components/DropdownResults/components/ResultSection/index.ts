export * from './components';
export * from './ResultSection';
