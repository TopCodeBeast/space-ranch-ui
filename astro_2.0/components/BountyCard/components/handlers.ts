import { SputnikNearService } from 'services/sputnik';

export const claimBountyHandler = (
  daoId: string,
  bountyId: string,
  deadline: string,
  bountyBond: string,
  onSuccess: () => void
) => (): void => {
  SputnikNearService.claimBounty(daoId, {
    bountyId: Number(bountyId),
    deadline,
    bountyBond,
  }).then(() => onSuccess());
};

export const unclaimBountyHandler = (
  daoId: string,
  bountyId: string,
  onSuccess: () => void
) => (): void => {
  SputnikNearService.unclaimBounty(daoId, bountyId).then(() => onSuccess());
};
