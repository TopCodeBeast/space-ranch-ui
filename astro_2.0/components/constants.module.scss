$nav-mobile-width: 920px;
