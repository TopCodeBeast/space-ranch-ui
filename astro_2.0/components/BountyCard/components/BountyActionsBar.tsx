import React from 'react';
import { InfoBlockWidget } from 'astro_2.0/components/InfoBlockWidget';
import { Button } from 'components/button/Button';
import { BountyStatus } from 'components/cards/bounty-card/types';
import { InfoValue } from 'astro_2.0/components/InfoBlockWidget/components/InfoValue';
import styles from './BountyActionsBar.module.scss';

interface BountyActionsBarProps {
  daoId: string;
  bountyId: string;
  bountyBond: string;
  forgivenessPeriod: string;
  bountyStatus: BountyStatus;
  currentUser: string;
  deadlineThreshold: string;
  proposalBond: string;
}

export const BountyActionsBar: React.FC<BountyActionsBarProps> = ({
  bountyStatus,
  bountyBond,
}) => {
  return (
    <div className={styles.root}>
      <InfoBlockWidget
        label="Bond"
        value={<InfoValue value={bountyBond} label="NEAR" />}
        tooltipText="You can unclaim a bounty for the next 2 days. If you miss your deadline, you will lose your bond."
        messageSeverity="Info"
      />
      <InfoBlockWidget
        label="Grace"
        value={<InfoValue value={bountyBond} label="Hours" />}
        tooltipText="You can unclaim a bounty for the next 2 days. If you miss your deadline, you will lose your bond."
        messageSeverity="Info"
      />
      {bountyStatus === 'Available' ? (
        <Button variant="black" size="medium">
          {/* onClick={proposeBountyDoneHandler()} */}
          Claim
        </Button>
      ) : (
        <>
          <Button
            variant="secondary"
            size="medium"
            // onClick={unclaimBountyHandler(daoId, bountyId)}
          >
            Unclaim
          </Button>

          <Button variant="black" size="small">
            {/* onClick={proposeBountyDoneHandler()} */}
            Purpose
          </Button>
        </>
      )}
    </div>
  );
};
