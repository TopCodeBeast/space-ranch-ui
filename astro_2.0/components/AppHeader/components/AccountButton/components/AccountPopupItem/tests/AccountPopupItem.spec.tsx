import { render } from 'jest/testUtils';
import { fireEvent } from '@testing-library/dom';

import { AccountPopupItem } from 'astro_2.0/components/AppHeader/components/AccountButton/components/AccountPopupItem';

describe('account popup item', () => {
  it('Should render component', () => {
    const { container } = render(<AccountPopupItem>Hey</AccountPopupItem>);

    expect(container).toMatchSnapshot();
  });

  it('Should call onClick', () => {
    const onClick = jest.fn();
    const component = render(
      <AccountPopupItem onClick={onClick}>Hey</AccountPopupItem>
    );

    fireEvent.click(component.getByRole('button'));

    expect(onClick).toBeCalled();
  });
});
