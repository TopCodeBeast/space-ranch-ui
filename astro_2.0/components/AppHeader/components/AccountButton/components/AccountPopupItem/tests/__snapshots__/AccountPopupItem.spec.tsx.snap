// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account popup item Should render component 1`] = `
<div>
  <div
    class=""
    role="button"
    tabindex="0"
  >
    Hey
  </div>
</div>
`;
